src/utils/liquidityMining.ts

```diff
@@ -0,0 +1,128 @@
+import Decimal from 'decimal.js-light'
+import {
+  CurrencyAmount,
+  LiquidityMiningCampaign,
+  Pair,
+  Percent,
+  Price,
+  USD,
+  ChainId,
+  Currency,
+  Token,
+  PricedToken,
+  PricedTokenAmount,
+  TokenAmount
+} from 'dxswap-sdk'
+import { getAddress, parseUnits } from 'ethers/lib/utils'
+import { SubgraphLiquidityMiningCampaign } from '../apollo'
+import { ZERO_USD } from '../constants'
+import { getLpTokenPrice } from './prices'
+
+export function getRemainingRewardsUSD(
+  campaign: LiquidityMiningCampaign,
+  nativeCurrencyUSDPrice: Price
+): CurrencyAmount {
+  const remainingRewards = campaign.remainingRewards
+  let remainingRewardsUSD = ZERO_USD
+  for (let i = 0; i < remainingRewards.length; i++) {
+    remainingRewardsUSD = remainingRewardsUSD.add(
+      CurrencyAmount.usd(
+        parseUnits(
+          remainingRewards[i].nativeCurrencyAmount.multiply(nativeCurrencyUSDPrice).toFixed(USD.decimals),
+          USD.decimals
+        ).toString()
+      )
+    )
+  }
+  return remainingRewardsUSD
+}
+
+export function getPairRemainingRewardsUSD(pair: Pair, nativeCurrencyUSDPrice: Price): CurrencyAmount {
+  // no liquidity mining campaigns check
```

Comment:
Might just be the wording, but I'm not quite sure what this mean. Is it checking that there are no liquidity mining campaigns?

Reply:
It basically gets all the remaining USD rewards for the given pair. There are 2 scenarios here:

- If no campaign are currently active on the pair, 0 is returned
- If one or multiple campaigns are currently active on the pair, for each one of them we calculate the remaining USD reward and we return the overall sum of it. The calculation is based on the amount of rewards in each campaign, start/end time and current time (to know how much of the rewards have already been given out and how many are in fact remaining, based on the progress percentage of the campaign), and the USD price of each reward token being distributed (as fetched from the subgraph).